### PR TITLE
chore: Use cli-runtime module to construct ClientConfig

### PR DESCRIPTION
**Pull request by danielpacak**
Adds support for standard `kubectl` options such as `--context`, `--as`, `--cluster`, etc.

Improves namespace handling, i.e.:

```
# In the current namespace if specified in the KUBECONF context, `default` otherwise:
kubectl who-can list pods

# All namespaces / cluster scope:
kubectl who-can list pods --all-namespaces

# In the namespace foo:
kubectl who-can list pods --namespace foo
```

### CODE REVIEW

File: cmd/kubectl-who-can.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import (
+	"github.com/aquasecurity/kubectl-who-can/pkg/cmd"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"os"
+)
+
+func main() {
+	root := cmd.NewCmdWhoCan()
+	if err := root.Execute(); err != nil {
+		os.Exit(1)
```
Comment (lizrice) [inline cmd/kubectl-who-can.go]: Should we be printing out the error?

File: pkg/cmd/list.go [hunk 2]
```diff
@@ -31,14 +34,98 @@ type whoCan struct {
...
+	var allNamespaces bool
+
+	configFlags := genericclioptions.NewConfigFlags(true)
+
+	cmd := &cobra.Command{
+		Use:   "kubectl-who-can VERB TYPE [NAME]",
+		Short: "who-can shows which users, groups and service accounts can perform a given action",
+		Long:  "who-can shows which users, groups and service accounts can perform a given verb on a given resource type",
+		Example: `  # List who can get pods in any namespace
+  kubectl who-can get pods
```
Comment (lizrice) [inline pkg/cmd/list.go]: I think this example is now listing pods in the current namespace? Might be nice to have an example for all namespaces as well, wdyt?

File: go.mod [hunk 3]
```diff
@@ -3,36 +3,12 @@ module github.com/aquasecurity/kubectl-who-can
 go 1.12
 
 require (
```
Comment (lizrice) [inline go.mod]: Just an observation, it's astonishing we can get rid of so many requirements!

File: cmd/kubectl-who-can.go (hunk 1 above)
Comment (danielpacak) [reply cmd/kubectl-who-can.go]: As far as I see it's already printed out as the first line in the output by cobra:

```
./kubectl-who-can get pod --xxx
Error: unknown flag: --xxx
Usage:
  ...
Examples:
  ...
Flags:
  ...
```

Printing it in the main function will output the error message twice:

```
./kubectl-who-can get pod --xxx
Error: unknown flag: --xxx
Usage:
  ...
Examples:
  ...
Flags:
  ...
Error: unknown flag: --xxx
```

File: pkg/cmd/list.go (hunk 2 above)
Comment (danielpacak) [reply pkg/cmd/list.go]: Good catch @lizrice . I've updated the examples.

**Pull request merged**